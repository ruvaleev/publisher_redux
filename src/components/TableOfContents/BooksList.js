import React from 'react';

import Book from '../Book';

function BooksList ({books, editable, chapterAdd, subsectionAdd}) {
  return (
    books.map((book, i) => (
      <Book
        key={`book_${book.id}`}
        book={book}
        editable={editable}
        chapterAdd={chapterAdd}
        subsectionAdd={subsectionAdd}/>
    ))
  )
}

export default BooksList;
