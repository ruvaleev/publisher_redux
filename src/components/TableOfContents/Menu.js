import React from 'react';
import { connect } from 'react-redux';

import { toggleEditable } from '../../redux/actions/tableOfContents';
import { showAll, showUncompleted, showCompleted } from '../../redux/actions/filters';

function Menu ({editable, filter, showAll, showUncompleted, showCompleted, toggleEditable, undo, redo}) {
  return (
    <div className='fixed -m-16 text-xl'>
      <button className='px-4' onClick={() => toggleEditable(editable)}>Edit</button>
      <FilterButton filter={filter} title='Show All' callback={showAll}/>
      <FilterButton filter={filter} title='Show Uncompleted' callback={showUncompleted}/>
      <FilterButton filter={filter} title='Show Completed' callback={showCompleted}/>
      <button className='px-4' onClick={() => undo()}>Undo</button>
      <button className='px-4' onClick={() => redo()}>Redo</button>
    </div>
  )
}

function FilterButton ({filter, title, callback}) {
  return (
    <button
      className={`px-4 rounded-lg border-gray-500 border-solid ${filter == title && 'border'}`}
      onClick={() => callback()}>
      {title}
    </button>
  )
}

const mapStateToProps = (state) => ({
  editable: state.tableOfContentsReducer.editable,
  filter: state.filtersReducer.title
})

const mapDispatchToProps = (dispatch) => ({
  toggleEditable: (editable) => dispatch(toggleEditable(editable)),
  showAll: () => dispatch(showAll()),
  showUncompleted: () => dispatch(showUncompleted()),
  showCompleted: () => dispatch(showCompleted()),
})

export default connect(mapStateToProps, mapDispatchToProps)(Menu);
