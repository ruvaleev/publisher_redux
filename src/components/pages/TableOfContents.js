import React from 'react';

const TableOfContents = (props) => {
  console.log(books);
  return (
    props.editable ?
      <>editable</>
    :
      <ContentsList/>
  );
}

const ContentsList = () => {
  return(
    books.map((book, i) => (
      <BookRow key={i} title={book.title} articles={book.articles}/>
    ))
  )
}

const BookRow = ({title, articles}) => {
  return (
    <tr className='mx-12'>
      <td className='p-2 text-xl'>{title}</td>
      {articles.map((article, i) => (
        <tr>
          <td>{article.title}</td>
          <td>
            <input type='checkbox' checked={article.ready}/>
          </td>
        </tr>
      ))}
    </tr>
  )
}

let books = [
  {
    title: "Тихий Дон",
    articles: [
      { title: 'глава 1', ready: false },
      { title: 'глава 2', ready: false },
      { title: 'глава 3', ready: true },
      { title: 'глава 4', ready: false }
    ]
  },
  {
    title: "Чипполино 2",
    articles: [
      { title: 'глава 1 - Первая кровь', ready: false },
      { title: 'глава 2 - Восставший из пепла', ready: true },
      { title: 'глава 3 - Кровавый закат', ready: false },
      { title: 'глава 4 - Торжество правосудия', ready: true }
    ]
  }
]

export default TableOfContents;
