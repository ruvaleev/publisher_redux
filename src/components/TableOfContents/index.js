import { connect } from 'react-redux';
import TableOfContents from './TableOfContents';

const mapStateToProps = (state) => ({
  isLoading: state.booksReducer.present.isLoading,
  books: state.booksReducer.present.books
})

export default connect(mapStateToProps)(TableOfContents);
