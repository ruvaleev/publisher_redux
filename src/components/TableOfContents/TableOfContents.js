import React from 'react';

import BooksList from './BooksList';
import Menu from './Menu';

function TableOfContents ({isLoading, isError, error, books}) {
  return (
    <>
      <Menu/>
      {
        isLoading
          ? <div>Loading...</div>
          : 
          isError
            ? <div>{error}</div>
            : <BooksList books={books}/>
      }
    </>
    )
}

export default TableOfContents;
