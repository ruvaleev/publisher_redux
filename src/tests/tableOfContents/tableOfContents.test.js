import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import TableOfContents from '../../components/TableOfContents';
import Store from '../shared/Store';

const mockStore = configureStore([]);

describe('TableOfContents', () => {
  let store;
  
  beforeEach(() => {
    store = mockStore(Store());
  });

  it('render correctly table of books', () => {
    const component = render(
      <Provider store={store}>
        <TableOfContents/>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
