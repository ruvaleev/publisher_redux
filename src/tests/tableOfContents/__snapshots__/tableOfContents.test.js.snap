// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableOfContents render correctly table of books 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="fixed -m-16 text-xl"
      >
        <button
          class="px-4"
        >
          Edit
        </button>
        <button
          class="px-4 rounded-lg border-gray-500 border-solid border"
        >
          Show All
        </button>
        <button
          class="px-4 rounded-lg border-gray-500 border-solid false"
        >
          Show Uncompleted
        </button>
        <button
          class="px-4 rounded-lg border-gray-500 border-solid false"
        >
          Show Completed
        </button>
        <button
          class="px-4"
        >
          Undo
        </button>
        <button
          class="px-4"
        >
          Redo
        </button>
      </div>
      <table
        class="mx-12"
      >
        <thead>
          <tr>
            <td
              class="p-2 text-xl"
            >
              Тихий Дон
            </td>
          </tr>
        </thead>
        <tbody
          class="flex flex-col"
        >
          <tr
            class="flex flex-col false"
          >
            <td
              class="flex"
            >
              глава 1
            </td>
            <td>
              <table>
                <tbody>
                  <tr>
                    <td
                      class="pl-4"
                    >
                      подраздел 1
                    </td>
                  </tr>
                </tbody>
              </table>
            </td>
          </tr>
          <tr />
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <div
      class="fixed -m-16 text-xl"
    >
      <button
        class="px-4"
      >
        Edit
      </button>
      <button
        class="px-4 rounded-lg border-gray-500 border-solid border"
      >
        Show All
      </button>
      <button
        class="px-4 rounded-lg border-gray-500 border-solid false"
      >
        Show Uncompleted
      </button>
      <button
        class="px-4 rounded-lg border-gray-500 border-solid false"
      >
        Show Completed
      </button>
      <button
        class="px-4"
      >
        Undo
      </button>
      <button
        class="px-4"
      >
        Redo
      </button>
    </div>
    <table
      class="mx-12"
    >
      <thead>
        <tr>
          <td
            class="p-2 text-xl"
          >
            Тихий Дон
          </td>
        </tr>
      </thead>
      <tbody
        class="flex flex-col"
      >
        <tr
          class="flex flex-col false"
        >
          <td
            class="flex"
          >
            глава 1
          </td>
          <td>
            <table>
              <tbody>
                <tr>
                  <td
                    class="pl-4"
                  >
                    подраздел 1
                  </td>
                </tr>
              </tbody>
            </table>
          </td>
        </tr>
        <tr />
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
